trio/tests/test_highlevel_open_tcp_stream.py

```diff
@@ -97,7 +97,7 @@ def test_format_host_port():
 # Now, thorough tests using fake sockets
 
 
-@attr.s
+@attr.s(cmp=False)
```

Comment:
I went with @njsmith's suggestion of a `set()` to track open sockets, but this requires that the socket is "hashable", which `attr` objects aren't by default

this seems to be the [suggested way](http://www.attrs.org/en/stable/hashing.html) of setting objects like this…

Reply:
:+1: